### PR TITLE
Create new integration if integrationId not provided

### PR DESCRIPTION
**Pull request by plxity**
Create new integration if integration ID is not provided is the `connectedAccounts.initiate` method.

----

> [!IMPORTANT]
> Add logic to create a new integration in `ConnectedAccounts.initiate` if `integrationId` is not provided.
> 
>   - **Behavior**:
>     - In `ConnectedAccounts.initiate`, create a new integration if `integrationId` is not provided and `authMode` is specified.
>     - Use `Apps.get` to retrieve app details and `Integrations.create` to create a new integration with a timestamped name.
>   - **Types**:
>     - `integrationId` in `InitiateConnectionDataReq` is now optional.
>   - **Misc**:
>     - Minor formatting changes in `connectedAccounts.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for c2ceb80549d3c265ab6779f12a494bd3ee4605cb. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: js/src/sdk/models/connectedAccounts.ts [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
 
 type InitiateConnectionDataReq = InitiateConnectionPayloadDto & {
     data?: Record<string, unknown> | unknown;
     entityId?: string;
     labels?: string[];
-    integrationId: string;
+    integrationId?: string;
     redirectUri?: string;
+    authMode?: string;
+    authConfig?: { [key: string]: any },
```
Comment (shreysingla11) [inline js/src/sdk/models/connectedAccounts.ts:19]: The `authConfig` type is using `any` which could lead to runtime errors. Consider defining a proper interface for better type safety:
```typescript
interface AuthConfig {
    clientId?: string;
    clientSecret?: string;
    [key: string]: string | undefined;
}
```

File: js/src/sdk/models/connectedAccounts.ts [hunk 2]
```diff
@@ -70,16 +79,31 @@ export class ConnectedAccounts {
...
-                data,
-            }  }).then(res => res.data);
-            
+            let { integrationId, entityId = 'default', labels, data = {}, redirectUri, authMode = '', authConfig = {}, appName } = payload;
+
+            if (!integrationId && authMode) {
+                const timestamp = new Date().toISOString().replace(/[-:.]/g, "");
+                const app = await this.apps.get({ appKey: appName! });
+                const integration = await this.integrations.create({
+                    appId: app.appId!,
```
Comment (shreysingla11) [inline js/src/sdk/models/connectedAccounts.ts:88]: Using non-null assertions (`!`) on `app.appId` and `integration.id` could lead to runtime errors if these values are undefined. Consider adding proper validation:
```typescript
if (!app?.appId) {
    throw new Error(`App not found or invalid app ID for ${appName}`);
}
if (!integration?.id) {
    throw new Error('Failed to create integration');
}
integrationId = integration.id;
```

File: js/src/sdk/models/connectedAccounts.ts [hunk 3]
```diff
@@ -70,16 +79,31 @@ export class ConnectedAccounts {
...
-      
-            const res = await client.connections.initiateConnection({ body: {
-                integrationId,
-                entityId,
-                labels,
-                redirectUri,
-                data,
-            }  }).then(res => res.data);
-            
+            let { integrationId, entityId = 'default', labels, data = {}, redirectUri, authMode = '', authConfig = {}, appName } = payload;
```
Comment (shreysingla11) [inline js/src/sdk/models/connectedAccounts.ts:82]: The initiate method should validate required parameters before proceeding. Consider adding input validation:
```typescript
if (!integrationId && !appName) {
    throw new Error('Either integrationId or appName must be provided');
}
if (!integrationId && !authMode) {
    throw new Error('authMode is required when creating new integration');
}
```

File: js/src/sdk/models/connectedAccounts.ts [hunk 4]
```diff
@@ -70,16 +79,31 @@ export class ConnectedAccounts {
...
+
+            if (!integrationId && authMode) {
```
Comment (shreysingla11) [inline js/src/sdk/models/connectedAccounts.ts:84]: Consider extracting the integration creation logic into a separate private method for better code organization and reusability:
```typescript
private async createNewIntegration(
    appName: string,
    authMode: string,
    authConfig: AuthConfig
): Promise<string> {
    const timestamp = new Date().toISOString().replace(/[-:.]/g, "");
    const app = await this.apps.get({ appKey: appName });
    if (!app?.appId) {
        throw new Error(`Invalid app: ${appName}`);
    }
    
    const integration = await this.integrations.create({
        appId: app.appId,
        name: `integration_${timestamp}`,
        authScheme: authMode,
        authConfig,
        useComposioAuth: false,
    });
    
    return integration?.id ?? throw new Error('Failed to create integration');
}
```

**Comment by shreysingla11** [comment]
### Code Review Summary

#### Overall Assessment
The changes introduce good functionality for dynamic integration creation, but there are several areas that need attention for better reliability and maintainability.

#### Strengths
✅ Good separation of concerns with proper class dependencies
✅ Flexible approach to handle both existing and new integrations
✅ Consistent error handling pattern using CEG.handleError

#### Areas for Improvement
1. **Type Safety**
   - Replace `any` type in authConfig with proper interface
   - Add proper null checks instead of non-null assertions

2. **Input Validation**
   - Add validation for required parameters
   - Validate appName and authMode when creating new integration

3. **Code Organization**
   - Extract integration creation logic to a separate method
   - Consider adding utility functions for common operations

4. **Documentation**
   - Add JSDoc comments for the modified methods
   - Document the new parameters and their requirements

#### Testing Requirements
- Unit tests for new integration creation flow
- Error case testing for invalid inputs
- Integration tests for the complete flow

The changes are generally good but implementing the suggested improvements would make the code more robust and maintainable.

Rating: 7/10 - Good functionality but needs improvements in type safety and validation.

File: js/src/sdk/models/connectedAccounts.ts (hunk 1 above)
Comment (plxity) [reply js/src/sdk/models/connectedAccounts.ts:19]: It's not required here.

File: js/src/sdk/models/connectedAccounts.ts (hunk 2 above)
Comment (plxity) [reply js/src/sdk/models/connectedAccounts.ts:93]: I've added it in the next commit

File: js/src/sdk/models/connectedAccounts.ts (hunk 4 above)
Comment (plxity) [reply js/src/sdk/models/connectedAccounts.ts:84]: Not required.

**Review by sohamganatra — APPROVED** [review]

**Pull request merged**